### PR TITLE
Support for load/stores for pointer types in worker-interp

### PR DESCRIPTION
**Pull request by nbushehri**
Also updated the some of the previous tests that included values larger than 255 for value bytes in memory

### CODE REVIEW

File: ir/instr.cpp [hunk 1]
```diff
@@ -3723,6 +3745,27 @@ static void storeIntVal(Interpreter &interpreter, util::ConcreteValPointer *ptr,
 
 }
 
+static void storePtrVal(Interpreter &interpreter, util::ConcreteValPointer *ptr_dst,
```
Comment (yotann) [inline ir/instr.cpp]: This function needs to store every byte of the pointer and set `ConcreteByte::pointer_byte_offset` to a different value for each byte. Then we can detect UB in cases like:

- Storing an 8-byte pointer to a 1-byte memory block.
- Storing a pointer that overwrites part of an integer, and then loading the integer.
- Storing an integer that overwrites part of a pointer, and then loading the pointer.

The number of bytes in a pointer depends on target architecture, looks like the correct number is `bits_program_pointer / bits_byte`.

File: ir/instr.cpp [hunk 2]
```diff
@@ -3618,6 +3618,29 @@ static util::ConcreteVal *loadIntVal(Interpreter &interpreter,
   return res;
 }
 
+static util::ConcreteVal *loadPtrVal(Interpreter &interpreter,
```
Comment (yotann) [inline ir/instr.cpp:3649]: Likewise needs to load all bytes of the pointer, and check that they all belong to the same pointer and have correct values in `ConcreteByte::pointer_byte_offset`.

File: ir/instr.cpp [hunk 3]
```diff
@@ -3630,7 +3653,12 @@ Load::concreteEval(Interpreter &interpreter) const {
     // todo handle multiple integer bitwidth
     return shared_ptr<ConcreteVal>(
         loadIntVal(interpreter, c_ptr, getType().bits()));
-  } else { // floating point, pointer, etc
+  }
+  else if (getType().isPtrType()) {
+    return shared_ptr<ConcreteVal>(
+        loadPtrVal(interpreter, c_ptr, getType().bits()));
```
Comment (yotann) [inline ir/instr.cpp]: `getType().bits()` is actually irrelevant for pointers. It's used for converting a pointer to a BV for the solver, but we aren't doing that, we're just storing the pointer directly in `ConcreteByte::ptr_val`.

**Review by yotann — CHANGES_REQUESTED** [review]

**Comment by nbushehri** [comment]
I think commit d6bb9aeeabd2720b70edf4d47729d6044e47d8e2 addresses the above issues.

**Comment by yotann** [comment]
I figured out why some bytes were greater than 255; see #3. I made some changes after accepting:

- 25cfa63ff34daec9382b4416c8c889f7cf452c3a and ebaee10a53e773d69827a90ec9c5f54ae0732a18: see #3 
- efa9ef3d41313af96f6d99697f0d95080984fae2: you forgot one check of is_ub
- 6af8b6fd15b90853781c26f893f18b771986092e: I fixed some asserts/UB that should have been UB/unsupported.

I also merged the upstream changes from Alive2, so you should merge my branch before your next PR.

**Pull request merged**